ner_v2/detectors/temporal/date/date_detection.py

```diff
@@ -0,0 +1,706 @@
+        super(DateDetector, self).__init__(language=language)
+
+    def detect_entity(self, text, **kwargs):
+        """
+        Detects all date strings in text and returns two lists of detected date entities and their corresponding
+        original substrings in text respectively.
+
+        Args:
+            text: string to extract date entities from
+
+        Returns:
+            Tuple containing two lists, first containing dictionaries, containing
+            date values as 'dd', 'mm', 'type', 'yy' for each detected date, and second list containing corresponding
+            substrings in given for entity detection
+
+            Examples:
+                date_detector = DateDetector("date")
+                date_detector.detect_entity('Remind me everyday')
+
+                date_detector.tagged_text
+
+
+        Additionally this function assigns these lists to self.date and self.original_date_text attributes
+        respectively.
+
+        """
+
+        self.text = ' ' + text.lower() + ' '
+        self.processed_text = self.text
+        self.tagged_text = self.text
+        if self.language_date_detector:
+            self.date, self.original_date_text = self.language_date_detector.detect_date(self.processed_text)
+        return self.date, self.original_date_text
+
+    def set_bot_message(self, bot_message):
+        """
+        Sets the object's bot_message attribute
+
+        Args:
+            bot_message: is the previous message that is sent by the bot
+        """
+        self.bot_message = bot_message
+
+    def to_datetime_object(self, base_date_value_dict):
+        """
+        Convert the given date value dict to a timezone localised datetime object
+
+        Args:
+            base_date_value_dict (dict): dict containing dd, mm, yy
+
+        Returns:
+            datetime object: datetime object localised with the timezone given on initialisation
+        """
+        datetime_object = datetime.datetime(year=base_date_value_dict['yy'],
+                                            month=base_date_value_dict['mm'],
+                                            day=base_date_value_dict['dd'], )
+        return self.timezone.localize(datetime_object)
+
+    @staticmethod
+    def to_date_dict(datetime_object, date_type=TYPE_EXACT):
+        """
+        Convert the given datetime object to a dictionary containing dd, mm, yy
+
+        Args:
+            datetime_object (datetime.datetime): datetime object
+            date_type (str, optional, default TYPE_EXACT): 'type' metdata for this detected date
+
+        Returns:
+            dict: dictionary containing day, month, year in keys dd, mm, yy respectively with date type as additional
+            metadata.
+        """
+        return {
+            'dd': datetime_object.day,
+            'mm': datetime_object.month,
+            'yy': datetime_object.year,
+            'type': date_type,
+        }
+
+    def _get_date_language_detector(self):
```

Comment:
This function need not exist. Combination of BaseDetector method (set_language_processing_script) and self.supported_languages  function should take care of it.

Reply:
done